### ISSUE TITLE
Inconsistent guidance on skip link placement

### ISSUE THREAD

**Issue by abbott567** [alphagov/govuk-design-system#2131]
## Related documentation
The [Skip Link Component](https://design-system.service.gov.uk/components/skip-link/)

## The problem
In the section titled: "When to use this component" the content reads:
> All GOV.UK pages must include a skip link in the header.

This implies that the Skip Link should be inside the `<header>` tag, but in the full page examples in the "patterns" section, such as the [question page example](https://design-system.service.gov.uk/patterns/question-pages/default/index.html) the skip link sits outside of all top level landmarks and is just an anchor tag on it's own at the top of the `<body>` tag.

We expect the correct implementation is for it to sit on its own outside of the header tag, but the documentation on the skip link component itself suggests otherwise.

## Issues witnessed
We have seen some teams who have perhaps taken the guidance on the Skip Link component literally, and putting the skip link inside the `<header>` tag, and this creates some weird behaviour on some screen reader features.

For example, on Voiceover if you use the option to go back to the start of the page, which is the following command, the screen reader jumps to the first item in the `<body>` tag.

<kbd>CONTROL</kbd> + <kbd>OPTION</kbd> + <kbd>FUNCTION </kbd> + <kbd>SHIFT</kbd> + <kbd>LEFT ARROW</kbd> 

On GOV.UK and the question page example, this is the Skip Link component. Because it's the first item, the skip-link takes the keyboard focus and the skip link appears visually at the top of the page due to its CSS focus styles. If you press tab from this point, the focus moves from the top of the page down the elements in order as you would expect it to.

However, if the skip link is inside of the `<header>` tag, when you use the screen reader functionality to jump back to the start of the page, it selects the header itself, so it just announces "banner." And, as the banner itself is not focusable, the keyboard focus stays where it was before you used the command. 

So, now if you press tab, unlike the previous example where the keyboard focus had been moved to the top, now, it remains where you left it on the page because the banner element did not take the focus with it due to it being an unfocusable element. If the user is used to the way GOVUK services work and expects them to be consistent, they may think they're at the top of the page, but they are continuing from part way down because the focus did not move.

## Suggestion
Make the guidance on where to put the skip link clear and align the patterns to use that guidance. So, whether you look at the skip link component in isolation, or you inspect the code in the pattern, there is no ambiguity for where the skip link should sit.

If the guidance is indeed to have the skip link on its own as the first element in the body, it would also be good to include guidance around the fact that Axe will flag it as a failure, because Axe expects all elements to be inside of a landmark. We know this to be a false flag, but we've seen teams wrap the skip link in `<nav>` tags and all sorts just to shut Axe up. 

Having strong guidance on why GDS built the skip link in the way that they have, will make it easier to explain to teams who are doing weird solutions to silence Axe failures.

**Comment by timpaul** [comment]
Thanks @abbott567 - this is a really detailed and helpful issue!

**Comment by EoinShaughnessy** [comment]
@abbott567 We've now published a fix for this. Thanks for taking the time to share those details, they were very helpful!
